src/main/java/org/embulk/output/sf_bulk_api/SForceColumnVisitor.java

```diff
@@ -21,37 +21,57 @@ public SForceColumnVisitor(SObject record, PageReader pageReader) {
 
   @Override
   public void booleanColumn(Column column) {
-    record.addField(column.getName(), pageReader.getBoolean(column));
+    if (pageReader.isNull(column)) {
+      record.addField(column.getName(), null);
+    } else {
+      record.addField(column.getName(), pageReader.getBoolean(column));
+    }
   }
 
   @Override
   public void longColumn(Column column) {
-    record.addField(column.getName(), Long.toString(pageReader.getLong(column)));
+    if (pageReader.isNull(column)) {
+      record.addField(column.getName(), null);
+    } else {
+      record.addField(column.getName(), Long.toString(pageReader.getLong(column)));
+    }
   }
 
   @Override
   public void doubleColumn(Column column) {
-    record.addField(column.getName(), Double.toString(pageReader.getDouble(column)));
+    if (pageReader.isNull(column)) {
+      record.addField(column.getName(), null);
+    } else {
+      record.addField(column.getName(), Double.toString(pageReader.getDouble(column)));
+    }
   }
 
   @Override
   public void stringColumn(Column column) {
-    record.addField(column.getName(), pageReader.getString(column));
+    if (pageReader.isNull(column)) {
+      record.addField(column.getName(), null);
+    } else {
+      record.addField(column.getName(), pageReader.getString(column));
+    }
   }
 
   @Override
   public void timestampColumn(Column column) {
-    Timestamp timestamp = pageReader.getTimestamp(column);
-    if (timestamp != null) {
+    if (pageReader.isNull(column)) {
+      record.addField(column.getName(), null);
+    } else {
+      Timestamp timestamp = pageReader.getTimestamp(column);
```

Comment:
If column is not null, Timestamp will never be null.
https://github.com/embulk/embulk/blob/v0.9.25/embulk-core/src/main/java/org/embulk/spi/PageReader.java#L106-L114